sql/core/src/test/scala/org/apache/spark/sql/SQLInsertTestSuite.scala

```diff
@@ -209,6 +211,43 @@ trait SQLInsertTestSuite extends QueryTest with SQLTestUtils {
     }
   }
 
+  test("SPARK-33474: Support typed literals as partition spec values") {
+    withTable("t1", "t2", "t4", "t5") {
+      val binaryStr = "Spark SQL"
+      val binaryHexStr = Hex.hex(UTF8String.fromString(binaryStr).getBytes).toString
+      sql("CREATE TABLE t1(name STRING, part DATE) USING PARQUET PARTITIONED BY (part)")
+      sql("INSERT INTO t1 PARTITION(part = date'2019-01-02') VALUES('a')")
```

Comment:
Can we create one table with many partition columns of different types, and test the positive normal cases together?
```
CREATE TABLE t1(name STRING, p1 DATE, p2 TIMESTAMP, p3 ...) USING PARQUET PARTITIONED BY (p1, p2, ...)
INSERT ...
```

Reply:
> Can we create one table with many partition columns of different types, and test the positive normal cases together?
> 
> ```
> CREATE TABLE t1(name STRING, p1 DATE, p2 TIMESTAMP, p3 ...) USING PARQUET PARTITIONED BY (p1, p2, ...)
> INSERT ...
> ```

Done, how about current?